### PR TITLE
Fix remaining add/edit command bugs 

### LINKED ISSUE #192: Alpha-bug: Add command name validation
Image

**Issues:**
* Add command allows contact with just `'`, `-`, or a combination of both as `n/NAME` to be added

**Suggestions**
* Perhaps we can restrict `n/NAME` to minimally have `x` amount of letters and must NOT consist of ONLY hyphens or apostrophes or combination of both.
   * However, even then, the above implementation is still vulnerable to other user's input. E.g. `n/abc-'-'-'-'-'-'-'-'`, `---abc----`, `'''abc'''`, `'a'b'c'''a'b'c'`, `a-b-c-b-b-b-b-a-a-a`, and other combinations alike
* Don't forget to make changes in UG too if applicable

### LINKED ISSUE #195: Alpha-bug: Add command email validation
Image

Image

Image

**Issues:**
* Add command allows addition of contact with `e/EMAIL` of format `x@xx` without any `.` which might not reflect real life emails correctly

**Suggestions:**
* Perhaps we improve the email validation logic to check that `e/EMAIL` must have at least one `.` in the domain labels
* [MINOR] Perhaps we can also improve the correct format guidance message to include correct email examples (i.e. which email is acceptable and which is not)
  * E.g. `xx@xx.xx is acceptable while xx@xx is not.` or something along this line

### LINKED ISSUE #201: Alpha-bug: Add command address validation
Image

**Issues:**
* Add command allows `a/ADDRESS` with symbols to be added
* Add command does not trim spaces in `a/ADDRESS` correctly

**Suggestions:**
* Perhaps we can improve the address validation to prevent symbols in `a/ADDRESS`
* Perhaps we can improve the address validation to trim any unnecessary spaces in `a/ADDRESS` 
* Don't forget to make changes in UG too if applicable

### LINKED ISSUE #206: Alpha-bug: Edit command update correct format guidance message
Image

**Issues:**
* [MINOR] Edit command correct format guidance message is missing `[b/BIRTHDAY]`

**Suggestions:**
* Let's add `[b/BIRTHDAY]` into the correct format guidance so user know birthday is modifiable

### LINKED ISSUE #207: Alpha-bug: Edit command invalid range message
Image

**Issues:**
* [MINOR] When trying to edit an out of range contact without any field provided, Edit command request for a field first

**Suggestions:**
* If possible, perhaps we can check whether the index is out of range first so user can fix their mistake even if they gave no fields

### PR DESCRIPTION
**Pull request by Hypovolemic**
image
image
image
image
image
image

* Closes #192 
* Closes #195 
* Closes #201
* Closes #206
* Closes #207

- Update UserGuide accordingly

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Name.java [hunk 1]
```diff
@@ -10,8 +10,9 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphabetic characters, spaces, hyphens, and apostrophes, "
-            + "and it should not be blank";
+            "Names should only contain alphabetic characters, spaces, hyphens, and apostrophes. "
+            + "Names must contain at least 2 alphabetic characters, cannot consist only of punctuation, "
+            + "and should not have consecutive punctuation (e.g., 'Mary-Jane' ✓, 'Mary--Jane' ✗).";
```
Comment (yalasoo) [inline src/main/java/seedu/address/model/person/Name.java]: Good handling of the consecutive punctuation case.

**Review by yalasoo — APPROVED** [review]
LGTM! Input validation is clear, edge cases are handled well also.

File: src/test/java/seedu/address/model/person/NameTest.java
Comment (TheresiaOng) [inline src/test/java/seedu/address/model/person/NameTest.java]: Great job on the extensive test cases!
